src/js/apis/c2c/DocumentService.js

```diff
@@ -7,6 +7,51 @@ DocumentService.prototype.getAll = function (params) {
   return this.api.get('/' + this.documentType + 's', { params });
 };
 
+DocumentService.prototype.fullDownload = function (params, limit, progress) {
+  // will load the ENTIRE list of document. Limited to 2000 docs
+  const this_ = this;
+
+  limit = limit || 2000;
+  params = { ...params }; // clone
```

Comment:
Note this is not a deep clone. For deep cloning, user `JSON.stringify`.
It's also a bad practive IMHO to assign the variable itself. I'd rather create a new one.

Reply:
It was on purpose, I didn't need a deep clone, only not modifying the original object when adding offset/limit.

By the way, I've changed it to a way more elegant version.